app/notification_push.go

```diff
@@ -404,11 +405,18 @@ func (a *App) SendAckToPushProxy(ack *model.PushNotificationAck) error {
 }
 
 func (a *App) getMobileAppSessions(userId string) ([]*model.Session, *model.AppError) {
-	result := <-a.Srv.Store.Session().GetSessionsWithActiveDeviceIds(userId)
-	if result.Err != nil {
-		return nil, result.Err
-	}
-	return result.Data.([]*model.Session), nil
+	sessionChan := make(chan store.StoreResult, 1)
```

Comment:
For this cases, where we are directly getting the data from the store with the `result := <-a.Srv.Store.Session().GetSessionsWithActiveDeviceIds(userId)`, this is working in a "sync" way, so we should use directly the sync approach, not the async one.

I would expect in this cases something like:
```go
sessions, err := a.Srv.Store.Session().GetSessionsWithActiveDeviceIds(userId)
if err != nil {
    return nil, err
}
return sessions, nil
```

But more concretely in this case, because the Store call and the API call is returning the same, we can just write

```go
return a.Srv.Store.Session().GetSessionsWithActiveDeviceIds(userId)
```

Reply:
Thank you for explaining this to me.